hoot-core/src/main/cpp/hoot/core/elements/Tags.cpp

```diff
@@ -235,6 +236,29 @@ QStringList Tags::getNames() const
   return result;
 }
 
+QStringList Tags::getNamesRecursive() const
+{
```

Comment:
How does this method differ from getNames()? It looks like you're limiting the names to only names that inherit from abstract_name. This should be a smaller list than getNames. Is that what you want?